pandas/core/series.py

```diff
@@ -988,7 +988,10 @@ def _set_with(self, key, value):
                 else:
                     return self._set_values(key, value)
             elif key_type == 'boolean':
-                self._set_values(key.astype(np.bool_), value)
+                try:
```

Comment:
Quite a few ways to do this but this was raising when any non-NumPy arrays were hitting this branch. Not directly related to change because it would fail for lists getting here but came to light as bitarrays were hitting this condition.